packages/openfeature-browser-provider/src/index.ts

```diff
@@ -100,66 +100,96 @@ export class BucketBrowserSDKProvider implements Provider {
     await this.initialize(newContext);
   }
 
-  resolveBooleanEvaluation(
+  private resolveFeature<T extends null | boolean | string | number | object>(
     flagKey: string,
-    defaultValue: boolean,
-  ): ResolutionDetails<boolean> {
-    if (!this._client)
+    defaultValue: T,
+  ): ResolutionDetails<T> {
+    const expType = typeof defaultValue;
+
+    if (!this._client) {
       return {
         value: defaultValue,
         reason: StandardResolutionReasons.DEFAULT,
         errorCode: ErrorCode.PROVIDER_NOT_READY,
-      } satisfies ResolutionDetails<boolean>;
+        errorMessage: "Bucket client not initialized",
+      } satisfies ResolutionDetails<T>;
+    }
 
     const features = this._client.getFeatures();
     if (flagKey in features) {
       const feature = this._client.getFeature(flagKey);
+
+      if (!feature.isEnabled) {
+        return {
+          value: defaultValue,
+          reason: StandardResolutionReasons.DISABLED,
+        };
+      }
```

Comment:
looks like this will return default value is the feature is not enabled? When fetching configs, i think we should just return the config value regardless of the `isEnabled` state. If they want this logic, they'll need to handle that in their application.